kernel/sched.c

```diff
@@ -1599,3 +1599,12 @@ int z_sched_waitq_walk(_wait_q_t  *wait_q,
 
 	return status;
 }
+
+#ifdef CONFIG_TEST
```

Comment:
Why the #if?  Normally you do this if it's a static function and will trigger an "unused symbol" warning if the code that calls it is removed by preprocessor state.  But this is a public symbol, it will build just fine and then be dropped at the link level.  Suggest removing for simplicity.